lib/loader/loader.c

```diff
@@ -238,3 +231,90 @@ struct evmc_instance* evmc_load_and_create(const char* filename,
 
     return instance;
 }
+
+struct evmc_instance* evmc_load_and_configure(const char* config,
+                                              enum evmc_loader_error_code* error_code)
+{
+    enum evmc_loader_error_code ec = EVMC_LOADER_SUCCESS;
+    struct evmc_instance* instance = NULL;
+
+    char config_copy_buffer[PATH_MAX_LENGTH];
+    if (strcpy_sx(config_copy_buffer, sizeof(config_copy_buffer), config) != 0)
+    {
+        ec = set_error(EVMC_LOADER_INVALID_ARGUMENT,
+                       "invalid argument: configuration is too long (maximum allowed length is %d)",
+                       (int)sizeof(config_copy_buffer));
+        goto exit;
+    }
+
+    char* options = "";
+    char* path_sep = strchr(config_copy_buffer, ',');
+    if (path_sep)
+    {
+        *path_sep = '\0';
+        options = path_sep + 1;
+    }
+    const char* path = config_copy_buffer;
+
+    instance = evmc_load_and_create(path, error_code);
+    if (!instance)
+        return NULL;
+
+    if (instance->set_option == NULL && strlen(options) != 0)
+    {
+        ec = set_error(EVMC_LOADER_INVALID_OPTION_NAME, "%s (%s) does not support any options",
+                       instance->name, path);
+        goto exit;
+    }
+
+
+    while (strlen(options) != 0)  // TODO: Can be optimized as *options != 0 because MSVC sucks.
+    {
+        char* option = options;
```

Comment:
Why not use `strtok` to do all these?

Reply:
Not thread safe.

Reply:
Well you can use `strtok_r`. But I wonder if a helper similar to `strtok` in this file could simplify the code?

Reply:
In my opinion it is not worth it. I fear it will be the same story as with `strcpy_s`.
- https://linux.die.net/man/3/strtok_r
- https://docs.microsoft.com/pl-pl/cpp/c-runtime-library/reference/strtok-s-strtok-s-l-wcstok-s-wcstok-s-l-mbstok-s-mbstok-s-l?view=vs-2019
- https://stackoverflow.com/questions/9021502/whats-the-difference-between-strtok-r-and-strtok-s-in-c

In practice anything from C standard lib besides `memcpy` and `strlen` is barely usable in portable way.

Do you want to try `strtok`? I might check it out our of curiosity.

Reply:
I mean you could have your own strtok-inspired helper in the code to be used in the 3 places where you tokenize. Not sure if it would simplify or just complicate the code.

Reply:
There is interesting behavior of `strtok`:

> A sequence of two or more contiguous delimiter bytes in the parsed string is considered to be a single delimiter. Delimiter bytes at the start or end of the string are ignored. Put another way: the tokens returned by strtok() are always nonempty strings. 

Should we follow this?

Reply:
Added `get_prefix()` helper, which is a bit different from `strtok()`. Code looks nice after changes.
I also added one more commit that optimizes `get_prefix()`, but I'm not sure this is the right change.

Reply:
I guess it is fine, considering with a comma delimited list of options, adding extra commas shouldn't do any harm. Question is do you want to allow that or raise an error?

Reply:
> I guess it is fine, considering with a comma delimited list of options, adding extra commas shouldn't do any harm. Question is do you want to allow that or raise an error?

After considerations, this introduces other edge cases, e.g. `option===value` would mean `option=value`. So I prefer leave it as it is.